shipwire/api.py

```diff
@@ -128,9 +139,34 @@ def _make_uri(self):
                                   method)
         return uri
 
+
 class ShipwireError(Exception):
     """
     Base Exception thrown by the Shipwire object when there is a
     general error interacting with the API.
     """
     pass
+
+
+class ResponseError(ShipwireError):
+    """
+    Exception raised when a response indicating failure is encountered
+    """
+
+    def __init__(self, response):
+        try:
+            message = response.json()['message']
+        except:
+            message = "Unknown message"
+
+        super(ResponseError, self).__init__(
+            'Unexpected Status Code (%d): %s' %
+            (response.status_code, message))
+        self.response = response
+
+
+class TimeoutError(ShipwireError):
+    """
+    Exception raised when a timeout occurs.
+    """
+    pass
```

Comment:
@clarkf since there's now more then just the generic error class can we move all of the errors to their own file like errors.py?